src/main/java/Duke.java

```diff
@@ -24,13 +27,18 @@ public void run() {
         sayGreeting();
         String command;
         Scanner sc = new Scanner(System.in);
-        while (true) {
+        do {
             command = sc.nextLine();
-            if (command.equals("bye")) {
+            switch (command) {
+            case "list":
+                printTasks();
+                break;
+            case "bye":
                 break;
+            default:
+                addTask(command);
             }
-            printFormatted(command);
-        }
+        } while (!command.equals("bye"));
         sayBye();
     }
 
```

Comment:
Splitting the different lines of codes in response to different commands makes your code really readable, which I really like!! Should have done this myself :)